server/plugin/api.go

```diff
@@ -314,11 +314,43 @@ func (p *Plugin) completeConnectUserToGitHub(c *Context, w http.ResponseWriter,
 		return
 	}
 
-	appErr = p.API.KVDelete(stateToken)
-	if appErr != nil {
-		p.API.LogWarn("Failed to delete state token", "error", appErr.Error())
-		http.Error(w, "error deleting stored state", http.StatusBadRequest)
-		return
+	if strings.Contains(stateToken, githubTokenKey) {
```

Comment:
Shouldn't this instead be something like:

```go
if !strings.Contains(stateToken, githubTokenKey) {
    // deliver error response
    return
}
```

Then we continue on with the success path

Reply:
@mickmister here we are accepting the state shouldn't have the `githubTokenKey` which is normal flow , if the state value consist of the `githubTokenKey` then this is an error case

Reply:
Okay so if the state has `githubTokenKey` then that's unwanted. So what _should_ we expect to be in `state`? Please take a look at the other related OAuth endpoints to see what we are setting  this value to.

Reply:
@mickmister currently insteady of direct deleting the state from KVStore , we are checking the state first and then the code too is verified , immediately after that it is deleted , hope this will work fine